src/common/controls/Control.js

```diff
@@ -1,11 +1,54 @@
-/* eslint-disable class-methods-use-this */
+import Observable from 'ol/Observable';
 
-class Control {
-  activate(map) {
+/**
+ * Generic control for mobility-toolbox-js.
+ */
+class Control extends Observable {
```

Comment:
What if we extend with the Control class of OpenLayers ?  and just redefining the setMap function. I think we could gain some functionnalities : get/set properties, auto trigger of events when a property change.